tests/test_git.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Pranjal Aswani <aswani.pranjal@gmail.com>
+
+
+import sys
+import unittest
+from datetime import datetime
+
+sys.path.insert(0, '..')
+
+import pandas as pd
+from numpy.testing import assert_array_equal
+from elasticsearch import Elasticsearch
+
+from manuscripts2.metrics import git
+from manuscripts2.elasticsearch import Query, Index, get_trend
+from utils import setup_data_source, teardown_data_source, DATA_SOURCES
+
+
+ES_URL = "http://localhost:9200"
+
+# All values as seen on 2018-07-10
+TREND_LAST = 12
+TREND_PRECENTAGE = -41
+
+# files
+AUTHORS_BY_PERIOD = "data/git_authors_by_months.csv"
+
+
+class TestGit(unittest.TestCase):
+    """
+    Test the git data source
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Setup necessary infrastructure to test the functions
+        """
+
+        cls.git = DATA_SOURCES['git']
+        setup_data_source('git', cls.git[0], cls.git[1])
+
+    def setUp(self):
+        """
+        Set up the necessary functions to run unittests
+        """
+
+        self.es = Elasticsearch(ES_URL)
+        self.git = DATA_SOURCES['git']
+        self.git_index = Index(self.git[1])
+
+        Query.interval_ = "month"
+
+        # Make sure to change the CONSTANTS defined above if you change
+        # the end date here before testing because tests might fail otherwise
+        self.start = datetime(2015, 1, 1)  # from date
+        self.end = datetime(2018, 7, 10)  # to date
+
+    def test_overview(self):
+        """
+        Test the overview function
+        """
+
+        overview = git.overview(self.git_index, self.start, self.end)
+
+        last, trend_percentage = get_trend(overview['activity_metrics'][0].timeseries())
+        self.assertEquals(last, TREND_LAST)
+        self.assertEquals(trend_percentage, TREND_PRECENTAGE)
+
+        authors = overview['author_metrics'][0].timeseries()
```

Comment:
If you're testing that timeseries is properly returning a data frame, that's ok. But if you just want to check that the timeseries is find, you can use the datastructure that does not imply returning a dataframe, and thus is more simple.

BTW, for checking that if you as a dataframe, you get a dataframe, I would write a separate check.

Reply:
I'll update the tests and separate the dataframe and non dataframe tests. Thanks!

Reply:
Thanks to you.